src/Tribe/Ignored_Events.php

```diff
@@ -0,0 +1,526 @@
+			add_filter( 'post_row_actions', array( $this, 'filter_actions' ), 10, 2 );
+
+			add_action( 'wp_ajax_tribe_convert_legacy_ignored_events', array( $this, 'ajax_convert_legacy_ignored_events' ) );
+
+			/**
+			 * Register Notices
+			 */
+			Tribe__Admin__Notices::instance()->register( 'legacy-ignored-events', array( $this, 'render_notice_legacy' ), 'dismiss=1&type=warning' );
+		}
+
+		public function render_notice_legacy() {
+			if ( ! Tribe__Admin__Helpers::instance()->is_post_type_screen( Tribe__Events__Main::POSTTYPE ) ) {
+				return false;
+			}
+
+			if ( empty( $_GET['post_status'] ) || $_GET['post_status'] !== self::$ignored_status ) {
+				return false;
+			}
+
+			if ( ! $this->has_legacy_deleted_posts() ) {
+				return false;
+			}
+
+			$html = '<p>' . '@TODO: Include a Cool message about why you are seen this Notice!' . '</p>';
+			$html .= '<p style="display:inline-block;">' . get_submit_button( esc_html__( 'Migrate Legacy Ignored Events' ), 'secondary', 'tribe-migrate-legacy-events', false ) . '<span class="spinner"></span>' . '</p>';
+
+			return Tribe__Admin__Notices::instance()->render( 'legacy-ignored-events', $html );
+		}
+
+		public function maybe_restore_events( $screen ) {
+			if ( ! isset( $_GET['action'] ) || 'tribe-restore' !== $_GET['action'] ) {
+				return;
+			}
+
+			$event = get_post( absint( $_GET['post'] ) );
+
+			if ( ! $event instanceof WP_Post ) {
+				return;
+			}
+
+			if ( Tribe__Events__Main::POSTTYPE !== $event->post_type ) {
+				return;
+			}
+
+			if ( self::$ignored_status !== $event->post_status ) {
+				return;
+			}
+
+			$sendback = remove_query_arg( array( 'trashed', 'untrashed', 'deleted', 'locked', 'ids' ), wp_get_referer() );
+
+			if ( isset( $_REQUEST['ids'] ) ) {
+				$post_ids = explode( ',', $_REQUEST['ids'] );
+			} elseif ( ! empty( $_REQUEST['post'] ) ) {
+				$post_ids = array_map( 'intval', (array) $_REQUEST['post'] );
+			}
+
+			$restored = 0;
+			foreach ( (array) $post_ids as $post_id ) {
+				if ( ! current_user_can( 'delete_post', $post_id ) ){
+					wp_die( __( 'You are not allowed to restore this item from the Ignored Events.', 'the-events-calendar' ) );
+				}
+
+				if ( ! $this->restore_event( $post_id ) ){
+					wp_die( __( 'Error in restoring from Ignored Events.' ) );
+				}
+
+				$restored++;
+			}
+			$sendback = add_query_arg( 'restored', $restored, $sendback );
+
+			wp_redirect( $sendback );
+			exit;
+		}
+
+		public function filter_actions( $actions, $post ) {
+			$event = get_post( $post );
+
+			if ( ! $event instanceof WP_Post ) {
+				return $actions;
+			}
+
+			if ( Tribe__Events__Main::POSTTYPE !== $event->post_type ) {
+				return $actions;
+			}
+
+			if ( self::$ignored_status !== $event->post_status ) {
+				return $actions;
+			}
+
+			$origin = get_post_meta( $event->ID, '_EventOrigin', true );
+
+			foreach ( $actions as $key => $html ) {
+				/**
+				 * @todo  Add logic to prevent removal of edit on EA origin
```

Comment:
Not sure if we'd need to resolve this `@todo` prior to merge or not ... can you confirm?